resources/js/components/ui/Modal/Modal.tsx

```diff
@@ -0,0 +1,53 @@
+import * as dialog from "@zag-js/dialog";
+import {normalizeProps, useMachine} from "@zag-js/solid";
+import {ParentComponent, Show, createComputed, createMemo, createUniqueId} from "solid-js";
+import {Portal} from "solid-js/web";
+import s from "./Modal.module.scss";
+
+interface Props {
+  title?: string;
+  open: boolean;
+  /**
+   * Event triggered when the Esc key is pressed or when the user clicks outside of the modal.
+   * If specified, it can react to that event by closing the dialog (i.e. setting the open prop
+   * to false).
+   */
+  onEscape?: (reason: "key" | "click") => void;
+}
+
+export const Modal: ParentComponent<Props> = (props) => {
+  const [state, send] = useMachine(
+    dialog.machine({
+      closeOnOutsideClick: false,
+      closeOnEsc: false,
+      onEsc: () => props.onEscape?.("key"),
+      onOutsideClick: () => props.onEscape?.("click"),
+      id: createUniqueId(),
+    }),
+  );
+  const api = createMemo(() => dialog.connect(state, send, normalizeProps));
+  createComputed(() => {
+    if (props.open) {
+      api().open();
+    } else {
+      api().close();
+    }
+  });
+  return (
+    <Show when={api().isOpen}>
+      <Portal>
```

Comment:
teraz jest ok, ale jak dojdą kolejne elementy, które będą używać Portala, to trzeba będzie utworzyć w template jakiegoś diva, którego referencje się poda do Portalu, aby wszystko lądowało do jednego miejsca

Reply:
Już teraz są dwa portale zdaje się. Nie tak całkiem rozumiem dlaczego by należało podawać ten mount inny niż domyślny, to może jak ktoś kiedyś będzie miał ochotę to machnie :)